### PR TITLE
Api functions

### PR DESCRIPTION
**Pull request by AJDurant**
updates to make the API more usable

### CODE REVIEW

File: src/Classes/ServiceAPI/MyRadio_Timeslot.php [hunk 1]
```diff
@@ -348,6 +348,17 @@ public static function getMostListened($date = 0)
         return $top;
     }
 
+    public static function getFromSlug($slug)
```
Comment (LloydW93) [inline src/Classes/ServiceAPI/MyRadio_Timeslot.php]: I kinda feel like there should be a more general purpose search-by-meta function. If there isn't one already. And if there isn't one already, how did we get this far? :P

Comment (AJDurant) [inline src/Classes/ServiceAPI/MyRadio_Timeslot.php]: yeah probably, and this is pretty dirty as I didn't even limit it to the correct metadata type

File: schema/api.json [hunk 2]
```diff
@@ -18,6 +18,8 @@
 	"\\MyRadio\\ServiceAPI\\MyRadio_Team": "team",
 	"\\MyRadio\\ServiceAPI\\MyRadio_TracklistItem": "tracklistItem",
 	"\\MyRadio\\ServiceAPI\\MyRadio_User": "user",
+    "\\MyRadio\\ServiceAPI\\MyRadio_Banner": "banner",
+    "\\MyRadio\\ServiceAPI\\MyRadio_BannerCampaign": "bannerCampaign",
```
Comment (LordAro) [inline schema/api.json]: muh tabs

Comment (AJDurant) [inline schema/api.json]: fixed

**Pull request merged**